packages/components/src/components/alertbox/alertbox.tsx

```diff
@@ -0,0 +1,176 @@
+import classNames from 'classnames';
+import statusNote from '../../utils/status-note';
+
+@Component({
+  tag: 'scale-alertbox',
+  styleUrl: 'alertbox.css',
+  shadow: true,
+})
+export class Alertbox {
+  // Color to variant rename
+  @Prop() variant?: 'informational' | 'success' | 'warning' | 'error' =
+    'informational';
+  @Prop({ reflect: true }) hasClose?: boolean = false;
+  @Prop({ reflect: true }) opened: boolean;
+  @Prop() timeout?: boolean | number = false;
+  /** (optional) aria-label attribute */
+  @Prop() ariaLabel: string;
+  /** (optional) aria-description attribute */
+  @Prop() ariaDescription: string;
+  @State() content: boolean = true;
+  @Element() hostElement: HTMLElement;
+
+  defaultTimeout = 3000;
+
+  componentDidLoad() {
+    this.content = !!this.hostElement.querySelector("p[slot='text']");
+    this.handleSlotAccessibility();
+  }
+
+  handleSlotAccessibility() {
+    const mainText =
+      this.content &&
+      this.hostElement.querySelector("p[slot='text']").innerHTML;
+    this.hostElement.shadowRoot
+      .querySelector('.alertbox__content')
+      .setAttribute('aria-description', mainText);
+
+    const headerText = this.hostElement.querySelector("p[slot='header']")
+      .innerHTML;
+    if (headerText) {
+      this.hostElement.shadowRoot
+        .querySelector('.alertbox__container-header')
+        .setAttribute('aria-label', headerText);
+    }
+  }
+
+  connectedCallback() {
+    statusNote({ source: this.hostElement, type: 'warn' });
+  }
+
+  /** Alert method: open() */
+  @Method()
+  async open() {
+    this.opened = true;
+  }
+
+  handleIcons() {
+    if (this.variant) {
+      switch (this.variant) {
+        case 'success':
+          return (
+            <scale-icon-action-success
+              class="alertbox__icon-success"
+              accessibility-title="success"
+            />
+          );
+        case 'informational':
+          return (
+            <scale-icon-alert-information
+              class="alertbox__icon-information"
+              accessibility-title="information"
+            />
+          );
+        case 'error':
+          return (
+            <scale-icon-alert-error
+              class="alertbox__icon-error"
+              accessibility-title="error"
+            />
+          );
+        case 'warning':
+          return (
+            <scale-icon-alert-warning
+              class="alertbox__icon-warning"
+              accessibility-title="warning"
+            />
+          );
+      }
+    }
+    return;
+  }
+
+  close = () => {
+    this.opened = false;
+  };
+
+  onCloseAlertWithTimeout = () => {
+    if (this.timeout !== false) {
+      if (typeof this.timeout === 'string' && !isNaN(this.timeout)) {
+        setTimeout(this.close, this.timeout);
+      } else {
+        setTimeout(this.close, this.defaultTimeout);
+      }
+    } else {
+      return null;
+    }
+  };
+
+  render() {
+    this.onCloseAlertWithTimeout();
```

Comment:
I would put this call in a lifecycle method, probably on [`componentDidLoad`](https://stenciljs.com/docs/component-lifecycle#componentdidload). It's a nice practice to avoid any logic inside the `render` function.